doc/rtd/development/packaging.rst

```diff
@@ -0,0 +1,105 @@
+this, cloud-init is not published to PyPI as it is not intended to be consumed
+as a pure-python package. Instead, cloud-init uses meson_ to build all
+dependent data files, system-initialization scripts and global configuration
+files.
+
+The build system used by ``cloud-init`` is `meson`_. ``Cloud-init`` is not
+published to pypi.org because it is not intended to be consumed directly as a
+pure-python library, SDK or API for managing cloud configuration.
+
+``Cloud-init`` correct behavior depends on the tight integration of
+system services performing :ref:`the early boot stages<boot_stages>` and
+system-wide configuration. Upstream uses `meson`_ build backend to support the
+distribution of all dependent data files, system-initialization scripts and
+global configuration files.
+
+Guidelines
+==========
+
+Build Dependencies
+------------------
+To use ``meson``, the following build dependencies should be available on the
+system:
+
+- ``python3``
+- ``meson >= 0.63.0``
+- ``pkgconf``
+- ``bash-completion``
+
+Additional dependencies for systemd environments:
+
+- ``systemd-devel``
+- ``udev``
+
+Local build procedure
+---------------------
+
+The custom ``init_system`` option affect paths where init scripts,
+configuration files, executables and data files are installed. The
+``init_system`` value defaults to ``systemd``.
+
+Other meson options used by this project are:
+
+- prefix
+- bindir
+- datadir
+- libexecdir
+- sysconfdir
+
+If the packaging system's meson option defaults are insufficient for your
+needs, they may be overridden with
+``meson setup -D<option_name>=<option_value>``.
+
+Steps to validate ``cloud-init`` package builds in a development environment:
+
+.. code-block:: bash
+
+   meson setup builddir -Dinit_system=systemd
+   meson compile -C builddir
+   meson test -C builddir -v
+   meson install -C builddir --destdir=testinstall
+   # List installed files
+   find builddir/testinstall/
+
+
+Test builds of RPMs or DEBs
+---------------------------
+One should first look to official downstream cloud images containing pre-built
+packages of``cloud-init`` for your prefered operating system. However, to ease
+the testing of local changes, development-quality DEB or RPM packages can be
+built with one of the following scripts:
+
+If your environment already has the build dependencies installed, the
+above templates may be used to build development-quality RPM or DEB
+packages:
+
+.. code-block:: bash
+
+   ./packages/brpm --distro=redhat  # or --distro=suse to build an RPM
+   ./packages/bddeb -S -d  # to build a DEB
+
+OR if LXC is present, the full package build can be run in a container:
+
+.. code-block:: bash
+
+   ./tools/run-container ubuntu-daily:plucky --package --keep
+   ./tools/run-container rockylinux/9 --package --keep
+
+
+.. note::
+
+   FreeBSD, NetBSD and OpenBSD meson support has not yet been added to :file:`tools/build-on-*bsd` or :file:`meson.build`.
```

Comment:
We have some dragonfly code too, but that's not supported either.

Regardless, I don't think that we need to enumerate - lets just say "the BSDs" and that should suffice.